cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowStatsSlaHttpHandler.java

```diff
@@ -0,0 +1,173 @@
+import co.cask.cdap.internal.app.store.WorkflowDataset;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.http.AbstractHttpHandler;
+import co.cask.http.HttpResponder;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+/**
+ *
+ */
+
+@Singleton
+@Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}")
+public class WorkflowStatsSlaHttpHandler extends AbstractHttpHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(WorkflowStatsSlaHttpHandler.class);
+  protected final Store store;
+  private final MetricStore metricStore;
+
+  @Inject
+  public WorkflowStatsSlaHttpHandler(MetricStore metricStore,
+                                Store store) {
+    this.metricStore = metricStore;
+    this.store = store;
+  }
+
+  @GET
+  @Path("apps/{app-id}/workflows/{workflow-id}/statistics")
+  public void workflowStats(HttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId,
+                            @PathParam("app-id") String appId,
+                            @PathParam("workflow-id") String workflowId,
+                            @QueryParam("start") String start,
+                            @QueryParam("end") String end,
+                            @QueryParam("percentile") List<Integer> percentiles) throws Exception {
+
+    long startTime = TimeMathParser.parseTimeInSeconds(start);
+    long endTime = TimeMathParser.parseTimeInSeconds(end);
+
+    if (endTime < startTime || startTime < 0 || endTime < 0) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, "Wrong start or end time.");
+      return;
+    }
+
+    List<WorkflowDataset.WorkflowRunRecord> workflowRunRecords;
+
+    Id.Program programId = Id.Program.from(namespaceId, appId, ProgramType.WORKFLOW, workflowId);
+    try {
+      try {
+        workflowRunRecords = store.getWorkflowRuns(programId, startTime, endTime);
+
+      } catch (IllegalArgumentException exception) {
+        responder.sendJson(HttpResponseStatus.BAD_REQUEST, "Bad request");
+        return;
+      }
+    } catch (SecurityException exception) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, "You're not allowed to view this");
+      return;
+    }
+
+    int count = workflowRunRecords.size();
+
+    for (int i : percentiles) {
+      if (i > 100) {
+        responder.sendJson(HttpResponseStatus.BAD_REQUEST, "Wrong percentile values");
```

Comment:
should be more descriptive. ``percentile value should be less than or equal to 100``